README.md

```diff
@@ -1 +1,37 @@
-# flink-sql-benchmark
\ No newline at end of file
+# flink-sql-benchmark
+
+## Generate test dataset
+
+- Step 1: Prepare your environment
+
+  Make sure you have Hadoop and Hive installed in your cluster. `gcc` is also needed to build the TPC-DS data generator.
+
+- Step 2: Build the data generator
+
+  Run `./tpcds-build.sh` to download and build the TPC-DS data generator.
+
+- Step 3: Generate TPC-DS dataset
+
+  Run `./tpcds-setup.sh <SCALE_FACTOR>` to generate dataset. The "scale factor" represents how much
+  data you will generate, which roughly translates to gigabytes. For example, `./tpcds-setup.sh 10`
+  will generate about 10GB data. Note that the scale factor must be **greater than 1**.
+
+  `tpcds-setup.sh` will launch a MapReduce job to generate the data in text format. By default,
```

Comment:
We can have separated script for each step, but I think it's tedious for users to remember and execute these scripts one by one. So I prefer to keep `tpcds-setup.sh` (which calls the separated scripts under the hood) and make it the recommended way for data generation. In case of failure, we should make sure the script is idempotent, so that users can safely retry it.